src/credentials/DeviceAttestationConstructor.cpp

```diff
@@ -54,110 +79,71 @@ CHIP_ERROR DeconstructAttestationElements(const ByteSpan & attestationElements,
-            case kTimestampTagId:
-                VerifyOrReturnError(lastContextTagId == kAttestationNonceTagId, CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_ELEMENT);
-                VerifyOrReturnError(timestampExists == false, CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_ELEMENT);
-                ReturnErrorOnFailure(tlvReader.Get(timestamp));
-                timestampExists = true;
-                break;
-            case kFirmwareInfoTagId:
-                VerifyOrReturnError(lastContextTagId == kTimestampTagId, CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_ELEMENT);
-                VerifyOrReturnError(firmwareInfoExists == false, CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_ELEMENT);
-                ReturnErrorOnFailure(tlvReader.GetByteView(firmwareInfo));
-                firmwareInfoExists = true;
-                break;
-            default:
-                return CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_ELEMENT;
-            }
-
-            lastContextTagId = TLV::TagNumFromTag(tag);
-        }
-        else if (TLV::IsProfileTag(tag))
-        {
-            // vendor fields
-            bool seenProfile = false;
-            uint16_t currentVendorId;
-            uint16_t currentProfileNum;
-
-            currentVendorId   = TLV::VendorIdFromTag(tag);
-            currentProfileNum = TLV::ProfileNumFromTag(tag);
-            if (!seenProfile)
-            {
-                seenProfile = true;
-                vendorId    = currentVendorId;
-                profileNum  = currentProfileNum;
-            }
-            else
-            {
-                // TODO: do not check for this - map vendorId and profileNum to each Vendor Reserved entry
-                // check that vendorId and profileNum match in every Vendor Reserved entry
-                VerifyOrReturnError(currentVendorId == vendorId && currentProfileNum == profileNum,
-                                    CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_ELEMENT);
-            }
-
-            ByteSpan vendorReservedEntry;
-            ReturnErrorOnFailure(tlvReader.GetByteView(vendorReservedEntry));
-            VerifyOrReturnError(vendorReservedIdx < vendorReservedArraySize, CHIP_ERROR_NO_MEMORY);
-            vendorReservedArray[vendorReservedIdx++] = vendorReservedEntry;
-        }
-        else
+        if (!TLV::IsContextTag(tag))
+            break;
+
+        switch (TLV::TagNumFromTag(tag))
         {
+        case kCertificationDeclarationTagId:
+            VerifyOrReturnError(lastContextTagId == UINT32_MAX, CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_ELEMENT);
+            ReturnErrorOnFailure(tlvReader.GetByteView(certificationDeclaration));
+            certificationDeclarationExists = true;
+            break;
+        case kAttestationNonceTagId:
+            VerifyOrReturnError(lastContextTagId == kCertificationDeclarationTagId, CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_ELEMENT);
+            VerifyOrReturnError(attestationNonceExists == false, CHIP_ERROR_IM_MALFORMED_COMMAND_DATA_ELEMENT);
```

Comment:
All of the `VerifyOrReturnError(xxxxExists == false, ...)` are superfluous in all sub-cases if we already check lastContextTagId